src/Google.Bigquery.V2/BigqueryClient.cs

```diff
@@ -0,0 +1,164 @@
+    /// </para>
+    /// <para>
+    /// All instance methods declared in this class are virtual, with an implementation which simply
+    /// throws <see cref="NotImplementedException"/>. All these methods are overridden in <see cref="BigqueryClientImpl"/>.
+    /// </para>
+    /// </remarks>
+    public abstract partial class BigqueryClient
+    {
+        /// <summary>
+        /// The underlying Bigquery service object used by this client.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="BigqueryClient"/> class only provides convenience operations built on top of
+        /// an existing service object. Any more complex operations which are not supported by this wrapper may wish
+        /// to use the same service object as the wrapper, in order to take advantage of its configuration (for authentication,
+        /// application naming etc).
+        /// </remarks>
+        public virtual BigqueryService Service { get; }
+
+        /// <summary>
+        /// The ID of the project this client operates on. All operations that require a
+        /// project to be specified implicitly supply this ID.
+        /// </summary>
+        public virtual string ProjectId { get; }
+
+        /// <summary>
+        /// Asynchronously creates a <see cref="BigqueryClient"/>, using application default credentials if
+        /// no credentials are specified.
+        /// </summary>
+        /// <remarks>
+        /// The credentials are scoped as necessary.
+        /// </remarks>
+        /// <param name="credential">Optional <see cref="GoogleCredential"/>.</param>
+        /// <returns>The task representing the created <see cref="BigqueryClient"/>.</returns>
+        public static async Task<BigqueryClient> CreateAsync(string projectId, GoogleCredential credential = null)
+            // If no credentials have been specified, we fetch them "properly asynchronously"
+            // to avoid the Task.Run in the synchronous call
+            => Create(
+                Preconditions.CheckNotNull(projectId, nameof(projectId)),
+                credential ?? await GoogleCredential.GetApplicationDefaultAsync().ConfigureAwait(false));
+
+        /// <summary>
+        /// Synchronously creates a <see cref="BigqueryClient"/>, using application default credentials if
+        /// no credentials are specified.
+        /// </summary>
+        /// <remarks>
+        /// The credentials are scoped as necessary.
+        /// </remarks>
+        /// <param name="credential">Optional <see cref="GoogleCredential"/>.</param>
+        /// <returns>The created <see cref="BigqueryClient"/>.</returns>
+        public static BigqueryClient Create(string projectId, GoogleCredential credential = null)
+        {
+            Preconditions.CheckNotNull(projectId, nameof(projectId));
+            credential = credential ?? Task.Run(() => GoogleCredential.GetApplicationDefaultAsync()).Result;
+            if (credential.IsCreateScopedRequired)
```

Comment:
Possibly split out the following code into a separate (private) method?
So Create[Async]() both call this new method when they've fetched the credentials (if that was required)?